### PR TITLE
Add GitHub Actions for building docs,  code-style checking, publishing to PyPI

### PR DESCRIPTION
**Pull request by andersy005**
- One of the workflows included here involves automatically publishing intake-stac to PyPI (after a release is created), the repo admin will need to add PYPI_USERNAME and PYPI_PASSWORD secrets in https://github.com/pangeo/intake-stac/settings/secrets. This workflow will be triggered whenever a release is created on Github.

- This PR also (hopefully) fixes the readthedocs documentation build. [All of the (December) latest builds failed](https://readthedocs.org/projects/intake-stac/builds/)

### CODE REVIEW

**Comment by jhamman** [comment]
@andersy005 - thanks for working on this. I would like to move entirely to github actions. @scottyhq has more experience with actions than I do so I'm going to ask him to provide the initial review here. I'll make a few inline comments as well.

File: requirements.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@ intake>=0.5.1
 intake-xarray>=0.3
 scikit-image>=0.15
 sat-stac>=0.1.3
+rasterio
```
Comment (jhamman) [inline requirements.txt]: I don't think this needs to be a hard dependency. I think users will often want rasterio but it should be possible to use intake-stac without it.

**Comment by jhamman** [comment]
@andersy005, can you merge in master. I made a few small bug fixes to dependencies and docs yesterday. I think that has caused a small conflict in `docs/requirements.txt`.

**Comment by jhamman** [comment]
> One of the workflows included here involves automatically publishing intake-stac to PyPI (after a release is created), the repo admin will need to add PYPI_USERNAME and PYPI_PASSWORD secrets in https://github.com/pangeo/intake-stac/settings/secrets. This workflow will be triggered whenever a release is created on Github.

This is done.

> This PR also (hopefully) fixes the readthedocs documentation build. All of the (December) latest builds failed

I fixed this in #39

**Comment by scottyhq** [comment]
This is great. Thanks @andersy005  I'm all for it! 

I am not so familiar with running black+flake8, but seems like a great idea to implement early on.

We should make sure the config for these various tools is compatible - or delete the config for tools no longer being used - isort? pycodestyle? (https://github.com/pangeo-data/intake-stac/blob/master/setup.cfg)

might as well delete .travis.yml with this pull request too?

See https://github.com/psf/black/issues/1205

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - id: flake8
+
+  # -   repo: https://github.com/asottile/seed-isort-config
+  #     rev: v1.9.4
+  #     hooks:
+  #     -   id: seed-isort-config
+  # -   repo: https://github.com/pre-commit/mirrors-isort
+  #     rev: v4.3.21
+  #     hooks:
+  #     -   id: isort
```
Comment (jhamman) [inline .pre-commit-config.yaml]: @andersy005 - do we want to uncomment this?

Comment (andersy005) [reply .pre-commit-config.yaml]: Fixed!

**Comment by andersy005** [comment]
> might as well delete .travis.yml with this pull request too?

I've added a job for running tests via GitHub workflows. One thing that is missing is the codecov upload. Today, Codecov exempts uploads originating from public project using Travis-CI, Circle CI, or AppVeyor CI from needing an upload token. This isn't the case for [GithHub workflows](https://github.com/codecov/codecov-action#usage)(at least for the time being).

@jhamman, @scottyhq , can any of you navigate to 
https://codecov.io/gh/pangeo-data/intake-stac/settings to get the upload token and add it to the https://github.com/pangeo/intake-stac/settings/secrets page as `CODECOV_TOKEN`? 

Once the codecov step is done, I will delete the `travis.yml` config

**Comment by jhamman** [comment]
> @jhamman, @scottyhq , can any of you navigate to
https://codecov.io/gh/pangeo-data/intake-stac/settings to get the upload token and add it to the https://github.com/pangeo/intake-stac/settings/secrets page as CODECOV_TOKEN?

Done.

**Comment by andersy005** [comment]
I think this is done. See https://github.com/andersy005/intake-stac/actions/runs/31139273 for more. Once approved, I will go ahead and delete the travis config.

**Pull request merged**